test/vt_flash/newton.jl

```diff
@@ -1,11 +1,39 @@
 @testset "newton.jl" begin
     booth(x) = (x[1] + 2*x[2] - 7)^2 + (2*x[1] + x[2] - 5)^2
-    ∇booth(g, x) = begin g .= Float64[10*x[1] + 8*x[2] - 34, 8*x[1] + 10*x[2] - 38]; g end
-    Hbooth(hess, x) = begin hess .= Float64[10 8; 8 10]; hess end
+    gradbooth!(g, x) = begin g .= Float64[10*x[1] + 8*x[2] - 34, 8*x[1] + 10*x[2] - 38]; g end
+    hessbooth!(hess, x) = begin hess .= Float64[10 8; 8 10]; hess end
 
+    xoptim = [1, 3]  # booth function minimum
     x = [4, 2]
-    result = CubicEoS.newton(booth, ∇booth, Hbooth, x)
-    @test result.converged
-    @test result.argument ≈ [1, 3]
-    @test booth(result.argument) ≈ 0 atol=1e-10
+
+    @testset "Stop by gradient" begin
+        ptrx = pointer(x)
+        xbefore = copy(x)
+        result = CubicEoS.newton(booth, gradbooth!, hessbooth!, x)
+        @testset "Sameness of argument" begin
+            @test pointer(x) == ptrx
+            @test x == xbefore
+        end
+
+        @testset "Converged test" begin
+            @test result.converged
+            @test result.argument ≈ xoptim
+            @test booth(result.argument) ≈ 0 atol=1e-10
+        end
+    end
```

Comment:
Refactor of previous test and explicit check of non-touching of `x` by `CubicEoS.newton`.

Reply:
What should `pointer(x) == ptrx` check? That `x` is not reallocated?

Reply:
Yes, though it may be overkill. In past, I used convert inside the solver and sometimes x passed to the solver was destroyed by it.